drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-api/src/main/java/org/drools/workbench/screens/scenariosimulation/model/Scenario.java

```diff
@@ -106,6 +106,17 @@ public FactMappingValue addOrUpdateMappingValue(FactIdentifier factIdentifier, E
         return factMappingValues.stream().filter(e -> e.getFactIdentifier().equals(factIdentifier)).collect(toList());
     }
 
+    public void setIndex(int index) {
```

Comment:
We discussed to have index only as UI element so I think we can remove completely `setIndex` and `getIndex` from `Scenario`. WDYT?
If you prefer to keep it I suggest to consider it not like a field but more like a property: `withIndex()` and `hasIndex()` with internal fixed value like `null`